server/app/services/grid_service_instance_deployer.rb

```diff
@@ -27,68 +23,37 @@ def deploy(node, instance_number, deploy_rev, creds = nil)
     false
   end
 
-  # @param [HostNode] node
-  # @param [String] instance_number
-  # @param [String] deploy_rev
-  def wait_for_service_to_start(node, instance_number, deploy_rev)
-    Timeout.timeout(30) do
-      sleep 0.5 until self.deployed_service_container_exists?(instance_number, deploy_rev)
+  # @param [GridServiceInstance] service_instance
+  def wait_for_service_to_start(service_instance)
+    Timeout.timeout(300) do
+      until service_instance.reload.state == 'running' do
```

Comment:
This needs to wait for the `service_instance.reload.deploy_rev >= deploy_rev`? The instance might already be running when we deploy it.